src/legacy/core_plugins/kibana/public/management/sections/objects/components/objects_table/components/flyout/flyout.js

```diff
@@ -99,24 +109,259 @@ class FlyoutUI extends Component {
+   */
+  async getConflictResolutions(objects) {
+    const resolutions = {};
+    for (const { type, id, title } of objects) {
+      const overwrite = await new Promise((resolve) => {
+        this.setState({
+          conflictingRecord: {
+            id,
+            type,
+            title,
+            done: resolve,
+          },
+        });
+      });
+      resolutions[`${type}:${id}`] = overwrite;
+      this.setState({ conflictingRecord: undefined });
+    }
+    return resolutions;
+  }
+
+  /**
+   * Map Import Failure To Retry Object
+   *
+   * Takes a single import failure and builds a retry object for the resolve import errors API. Returns undefined when failure should be skipped.
+   */
+  mapImportFailureToRetryObject = ({ failure, overwriteDecisionCache, replaceReferencesCache }) => {
+    const { isOverwriteAllChecked, unmatchedReferences } = this.state;
+    const isOverwriteGranted = isOverwriteAllChecked || overwriteDecisionCache[`${failure.obj.type}:${failure.obj.id}`] === true;
+
+    // Conflicts wihtout overwrite granted are skipped
+    if (!isOverwriteGranted && failure.error.type === 'conflict') {
+      return;
+    }
+
+    // Replace references if user chose a new reference
+    if (failure.error.type === 'missing_references') {
+      const objReplaceReferences = replaceReferencesCache.get(`${failure.obj.type}:${failure.obj.id}`) || [];
+      const indexPatternRefs = failure.error.references.filter(obj => obj.type === 'index-pattern');
+      for (const reference of indexPatternRefs) {
+        for (const unmatchedReference of unmatchedReferences) {
+          const hasNewValue = !!unmatchedReference.newIndexPatternId;
+          const matchesIndexPatternId = unmatchedReference.existingIndexPatternId === reference.id;
+          if (!hasNewValue || !matchesIndexPatternId) {
+            continue;
+          }
+          objReplaceReferences.push({
+            type: 'index-pattern',
+            from: unmatchedReference.existingIndexPatternId,
+            to: unmatchedReference.newIndexPatternId,
+          });
+        }
+      }
+      replaceReferencesCache.set(`${failure.obj.type}:${failure.obj.id}`, objReplaceReferences);
+      // Skip if nothing to replace, the UI option selected would be --Skip Import--
+      if (objReplaceReferences.length === 0) {
+        return;
+      }
+    }
+
+    return {
+      id: failure.obj.id,
+      type: failure.obj.type,
+      overwrite: isOverwriteAllChecked || overwriteDecisionCache[`${failure.obj.type}:${failure.obj.id}`] === true,
+      replaceReferences: replaceReferencesCache.get(`${failure.obj.type}:${failure.obj.id}`) || [],
+    };
+  }
+
+  /**
+   * Resolve Import Errors
+   *
+   * Function goes through the failedImports and tries to resolve the issues.
+   */
+  resolveImportErrors = async () => {
+    const { intl } = this.props;
+    let overwriteDecisionCache = {};
+    const replaceReferencesCache = new Map();
```

Comment:
nit: it's a little unexpected that one of these is a Map and the other a plain object. They seemed to be used similarly.